src/py42/modules/securitydata.py

```diff
@@ -169,6 +172,65 @@ def search_file_events(self, query):
         file_event_client = self._microservices_client_factory.get_file_event_client()
         return file_event_client.search(query)
 
+    def _search_by_hash(self, hash, type):
+        query = FileEventQuery.all(type.eq(hash))
+        response = self.search_file_events(query)
+        return response["fileEvents"]
+
+    def _find_file_versions(self, md5_hash, sha256_hash):
+        file_event_client = self._microservices_client_factory.get_file_event_client()
+        pds_client = self._microservices_client_factory.get_preservation_data_service_client()
+        for device_id, paths in file_event_client.get_file_location_detail_by_sha256(sha256_hash):
+            yield pds_client.find_file_versions(md5_hash, sha256_hash, device_id, paths)
+
+    def _stream_file(self, file_generator, file):
+        for response in file_generator:
+            if response.status_code != 200:
+                continue
+            session = self._microservices_client_factory.get_storage_node_session(
+                response[u"storageNodeURL"]
+            )
+            storage_node_client = StorageNodeClient(
```

Comment:
Do we need to create `StorageNodeClient` in `client_factories` ?

Reply:
Probably, then we also wouldn't need `get_storage_node_session()`

Reply:
Yes, agreed. The api methods on storage nodes are not really microservices, so a `create_storage_preservation_client` method there would be ideal.

Reply:
done.